### PR TITLE
updated learn-intermediate-algorithmic-thinking-by-building-a-dice-game

### PR DESCRIPTION
**Pull request by JoyShaheb**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

### CODE REVIEW

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js [hunk 1]
```diff
@@ -1,272 +1,259 @@
...
-        this.validScoreOptions["full-house"] = 25;
-      }
-    }
+  if (smallStraightLogic.includes(stringifyArray)) {
+    updateRadioOption(3, 30);
+  } else if (largeStraightLogic.includes(stringifyArray)) {
+    updateRadioOption(4, 40);
+  } else {
+    updateRadioOption(5, 0);
+  }
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:124]:
```suggestion
  if (smallStraightLogic.includes(stringifyArray)) {
    updateRadioOption(3, 30);
  }
  if (largeStraightLogic.includes(stringifyArray)) {
    updateRadioOption(4, 40);
  }
  updateRadioOption(5, 0);
```

These should not be mutually exclusive, IMHO.

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js [hunk 2]
```diff
@@ -1,272 +1,259 @@
...
-    alert("Please select a score.");
-    return false;
+  // Return the highest duplicates only if there are at least three or four of them
+  if (highestCount >= 4) {
+    updateRadioOption(1, sumOfAllDices);
+  } else if (highestCount >= 3) {
+    updateRadioOption(0, sumOfAllDices);
+  } else {
+    updateRadioOption(5, 0);
   }
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:168]:
```suggestion
  if (highestCount >= 4) {
    updateRadioOption(1, sumOfAllDices);
  }
  if (highestCount >= 3) {
    updateRadioOption(0, sumOfAllDices);
  }
  updateRadioOption(5, 0);
```

These should not be mutually exclusive, IMHO.

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js [hunk 3]
```diff
@@ -1,272 +1,259 @@
...
+    }
+    if (count >= 4 && count > highestCount) {
+      highestCount = count;
     }
   }
 
-  isKeepScoreSuccess() {
-    const allScoreInputs = scoreInputs;
+  // Filter the original array to include only the numbers that appear the highest number of times
+  const highestDuplicates = arr.filter((num) => counts[num] === highestCount);
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:156]:
```suggestion
```

This does not appear to be used.

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js [hunk 4]
```diff
@@ -1,272 +1,259 @@
...
-  if (game.currentRound <= 6) {
-    game.rollDice();
-    game.resetRadioInputs();
-    game.generateValidScoreOptions();
-    game.enableValidScoreInputs();
-    game.updateStatsUI();
-    keepScoreBtn.disabled = false;
+  // Check if there are three of one number and two of another number
+  let hasThreeOfAKind = false;
+  let hasPair = false;
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:186]:
```suggestion
  // Check if there are three of one number and two of another number
  // Have campers iterate through object first, then refactor this to 
  // use Object.values().includes() instead.
  let hasThreeOfAKind = false;
  let hasPair = false;
```

A note for a cool refactor step.

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js [hunk 5]
```diff
@@ -1,272 +1,259 @@
...
-    game.resetRadioInputs();
-    game.updateStatsUI();
-    game.updateScoreUI();
-    rollDiceBtn.disabled = false;
-    e.target.disabled = true;
+  if (hasThreeOfAKind && hasPair) {
+    updateRadioOption(2, 25);
+  } else {
+    updateRadioOption(5, 0);
   }
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:200]:
```suggestion
  if (hasThreeOfAKind && hasPair) {
    updateRadioOption(2, 25);
  };
  updateRadioOption(5, 0);
```

These shouldn't be mutually exclusive IMHO.

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js [hunk 6]
```diff
@@ -1,272 +1,259 @@
...
+  resetRadioOption();
+};
+
+/**
+ * ! this is the trigger for the app to work
+ */
+
+rollDiceBtn.addEventListener("click", () => {
+  if (rolls === 3) {
+    window.alert("Please select an option");
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:243]:
```suggestion
    window.alert("You have made three rolls this round. Please select a score.");
```

Clarify the message.

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js [hunk 7]
```diff
@@ -1,272 +1,259 @@
...
 
-// Return the maximum number of consecutives
+/**
+ * ! this is the function that updates the rolls and rounds
+ */
 
-const getMaxNumOfConsecutives = (sortedArr) => {
-  let currentCount = 0;
-  let maxCount = 0;
+const UpdateStats = () => {
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:55]:
```suggestion
const updateStats = () => {
```

Consistency.

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js [hunk 8]
```diff
@@ -1,272 +1,259 @@
...
+ */
+
+rollDiceBtn.addEventListener("click", () => {
+  if (rolls === 3) {
+    window.alert("Please select an option");
+  } else {
+    rolls++;
+    resetRadioOption();
+    rollDice();
+    UpdateStats();
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:248]:
```suggestion
    updateStats();
```

Consistency.

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js [hunk 9]
```diff
@@ -277,3 +264,36 @@ rulesBtn.addEventListener("click", () => {
...
+      selectedValue = radioButton.value;
+      achieved = radioButton.id;
+      break;
+    }
+  }
+
+  if (selectedValue) {
+    rolls = 0;
+    round++;
+    UpdateStats();
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:287]:
```suggestion
    updateStats();
```
Consistency.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
This is looking good. Some quick changes.

---

Unrelated, but I do wonder if this is too similar to the debugging dice game. We may want to discuss this in chat.

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.html [hunk 10]
```diff
@@ -1,140 +1,106 @@
...
-          <strong>Round:</strong> <span id="current-round">1</span>
-        </p>
+  <main>
+    <form id="game">
+      <div id="dice">
+        <button class="die" type="button"></button>
+        <button class="die" type="button"></button>
+        <button class="die" type="button"></button>
+        <button class="die" type="button"></button>
+        <button class="die" type="button"></button>
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.html:54]: We shouldn't use buttons here if they're not interactable.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.html (hunk 10 above)
Comment (JoyShaheb) [reply javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.html]: changes made @naomi-lgbt

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js [hunk 11]
```diff
@@ -1,272 +1,265 @@
...
+  const counts = {};
 
-    if (totalConsecutives === 5) {
-      this.validScoreOptions["large-straight"] = 40;
+  // Count the occurrences of each number in the array
+  for (const num of arr) {
+    if (counts[num]) {
+      counts[num]++;
+    } else {
+      counts[num] = 1;
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:139]:
```suggestion
  // Count the occurrences of each number in the array
  // can have campers refactor to:
  // counts[num] = (counts[num] || 0) + 1;
  for (const num of arr) {
    if (counts[num]) {
      counts[num]++;
    } else {
      counts[num] = 1;
```

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js [hunk 12]
```diff
@@ -1,272 +1,265 @@
+// targets the radio buttons and the spans
 const scoreInputs = document.querySelectorAll("#score-options input");
 const scoreSpans = document.querySelectorAll("#score-options span");
-const diceDivs = document.querySelectorAll("#dice > div");
+
+// square boxes that display the dice values
+const diceButtons = document.querySelectorAll("#dice > button");
```
Comment (naomi-lgbt) [inline javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:6]:
```suggestion
const diceButtons = document.querySelectorAll(".die");
```

Updated query selector to reflect new HTML.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js (hunk 12 above)
Comment (JoyShaheb) [reply javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:6]: added

File: javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js [hunk 13]
```diff
@@ -277,3 +270,36 @@ rulesBtn.addEventListener("click", () => {
...
+    }
+  }
+
+  if (selectedValue) {
+    rolls = 0;
+    round++;
+    updateStats();
+    resetRadioOption();
+    updateScore(selectedValue, achieved);
+    if (round >= 6) {
```
[the remark this answers is not in the document]
Comment (JoyShaheb) [reply javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js:296]: updated

**Review by jdwilkin4 — APPROVED** [review]
I think these changes look good.
Let's get this merged in 👍

**Pull request merged**